lib/services/local/generic/generic_wrapper.go

```diff
@@ -125,14 +125,16 @@ func (s ServiceWrapper[T]) UpsertResource(ctx context.Context, resource T) (T, e
 	return adapter.resource, trace.Wrap(err)
 }
 
-// UpdateResource updates an existing resource.
-func (s ServiceWrapper[T]) UpdateResource(ctx context.Context, resource T) (T, error) {
+// UnconditionalUpdateResource updates an existing resource without checking the provided resource revision.
+// This is faster and cheaper than ConditionalUpdateResource but can cause data loss on concurrent writes.
+func (s ServiceWrapper[T]) UnconditionalUpdateResource(ctx context.Context, resource T) (T, error) {
 	adapter, err := s.service.UpdateResource(ctx, newResourceMetadataAdapter(resource))
 	return adapter.resource, trace.Wrap(err)
 }
 
 // ConditionalUpdateResource updates an existing resource if the provided
 // resource and the existing resource have matching revisions.
+// This is safer than UnconditionalUpdateResource but more expensive and might not suit high-load scenarios.
```

Comment:
More expensive than what? In a lot of backends the cost of the storage operation itself is pretty much the same, it's the usage pattern around optimistic locking that's more expensive than blindly upserting a value.

Without expanding too much in the docs here (unless you think it's useful) we could just link to https://github.com/gravitational/teleport/blob/master/rfd/0126-backend-migrations.md#optimistic-locking and https://github.com/gravitational/teleport/blob/master/rfd/0153-resource-guidelines.md#update-1 . WDYT?